pkg/scheduler/backend/queue/active_queue.go

```diff
@@ -331,15 +358,17 @@ func (aq *activeQueue) done(pod types.UID) {
 	aq.lock.Lock()
 	defer aq.lock.Unlock()
 
-	inFlightPod, ok := aq.inFlightPods[pod]
+	inFlightPods, ok := aq.inFlightPods[pod]
 	if !ok {
 		// This Pod is already done()ed.
 		return
 	}
 	delete(aq.inFlightPods, pod)
 
 	// Remove the pod from the list.
-	aq.inFlightEvents.Remove(inFlightPod)
+	for _, inFlightPod := range inFlightPods {
+		aq.inFlightEvents.Remove(inFlightPod)
```

Comment:
This seems a bit problematic. If a Pod is inflight more than once, does it make sense for the first `Done` to get rid of all of the events associated with it?

Maybe we should cut the problem when inserting the Pod into the inflight map.